### PR TITLE
[Fixing]Switch to new file selection component

### PR DESCRIPTION
**Pull request by Flanker32**
Switch to new file selection component, adapt with `AzureArtifact`

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/component/input/ComboBoxAzureArtifact.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+            ComboBoxAzureArtifact.this.setSelectedItem(selectArtifact);
+        } else {
+            ComboBoxAzureArtifact.this.setSelectedItem(existingArtifact);
+        }
+    }
+
+    private void resetDefaultValue(final AzureArtifactType defaultArtifactType, final String defaultPath) {
+        final List<AzureArtifact> artifacts = UIUtils.listComboBoxItems(this);
+        final AzureArtifact defaultArtifact =
+                artifacts.stream().filter(artifact -> StringUtils.equals(artifact.getTargetPath(), defaultPath)
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/component/input/ComboBoxAzureArtifact.java:133]: AzureArtifact is designed to use getArtifactIdentifier to detect same object,  see PR https://github.com/microsoft/azure-tools-for-java/pull/4655/files

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/common/AzureArtifactComboBox.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+    private void addOrSelectExistingVirtualFile(VirtualFile virtualFile) {
+        final AzureArtifact selectArtifact = AzureArtifact.createFromFile(virtualFile);
+        final List<AzureArtifact> artifacts = UIUtils.listComboBoxItems(AzureArtifactComboBox.this);
+        final AzureArtifact existingArtifact =
+                artifacts.stream().filter(artifact -> artifact.getType() == AzureArtifactType.Artifact
+                        && StringUtils.equalsAnyIgnoreCase(artifact.getTargetPath(), selectArtifact.getTargetPath()))
+                         .findFirst().orElse(null);
+        if (existingArtifact == null) {
+            AzureArtifactComboBox.this.addItem(selectArtifact);
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/common/AzureArtifactComboBox.java:123]: redundant `AzureArtifactComboBox.`

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/common/AzureArtifactComboBox.java [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+            AzureArtifactComboBox.this.setSelectedItem(selectArtifact);
+        } else {
+            AzureArtifactComboBox.this.setSelectedItem(existingArtifact);
+        }
+    }
+
+    private void resetDefaultValue(final AzureArtifactType defaultArtifactType, final String defaultPath, final String artifactIdentifier) {
+        final List<AzureArtifact> artifacts = UIUtils.listComboBoxItems(this);
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/common/AzureArtifactComboBox.java:131]: this.getItems()

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/common/AzureArtifactComboBox.java [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    public void setFileChooserDescriptor(final Condition<? super VirtualFile> filter) {
+        this.filter = filter;
+    }
+
+    public synchronized void refreshItems(AzureArtifactType defaultArtifactType, String defaultPath, String artifactIdentifier) {
+        unsubscribeSubscription(subscription);
+        this.setLoading(true);
+        subscription = this.loadItemsAsync()
+                           .subscribe(items -> DefaultLoader.getIdeHelper().invokeLater(() -> {
+                               this.removeAllItems();
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/common/AzureArtifactComboBox.java:68]: Need to keep current selection unchanged if it is still in items.

**Review by andxu — APPROVED** [review]

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/common/AzureArtifactComboBox.java [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+    private Condition<? super VirtualFile> filter;
+    private Subscription subscription;
+
+    public AzureArtifactComboBox(Project project) {
+        super(false);
+        this.project = project;
+    }
+
+    public void setFileChooserDescriptor(final Condition<? super VirtualFile> filter) {
+        this.filter = filter;
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/common/AzureArtifactComboBox.java:60]: how about rename to `fileChooserDescriptor`?

Comment (Flanker32) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/common/AzureArtifactComboBox.java:60]: Will rename it in next pr

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/common/AzureArtifactComboBox.java (hunk 4 above)
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/common/AzureArtifactComboBox.java:68]: `setItems` will of course remove all the existing items. so no need to remove here.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/common/AzureArtifactComboBox.java (hunk 5 above)
Comment (wangmingliang-ms) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/common/AzureArtifactComboBox.java:60]: seems should be `fileFilter`/`setFileFilter` according to its usage.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/common/AzureArtifactComboBox.java [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+        final AzureArtifact existingArtifact =
+                artifacts.stream().filter(artifact -> artifact.getType() == AzureArtifactType.Artifact
+                        && StringUtils.equalsAnyIgnoreCase(artifact.getTargetPath(), selectArtifact.getTargetPath()))
+                         .findFirst().orElse(null);
+        if (existingArtifact == null) {
+            this.addItem(selectArtifact);
+            this.setSelectedItem(selectArtifact);
+        } else {
+            this.setSelectedItem(existingArtifact);
+        }
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/common/AzureArtifactComboBox.java:127]:
```
ARTIFACT = ...
if ARTIFACT not in the list
   add ARTIFACT into the list first
   and then select ARTIFACT in the combobox
else
   select the ARTIFACT in the combobox directly
```
right?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/common/AzureArtifactComboBox.java [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+    }
+
+    private void resetDefaultValue(final AzureArtifactType defaultArtifactType, final String defaultPath, final String artifactIdentifier) {
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/common/AzureArtifactComboBox.java:130]: seems very similar to super's `setValue` method.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/common/AzureArtifactComboBox.java (hunk 6 above)
Comment (Flanker32) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/common/AzureArtifactComboBox.java:127]: Will refactor in later PR

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/common/AzureArtifactComboBox.java (hunk 7 above)
Comment (Flanker32) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/common/AzureArtifactComboBox.java:130]: Will refactor in later PR

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/common/AzureArtifactComboBox.java (hunk 4 above)
Comment (Flanker32) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/common/AzureArtifactComboBox.java:68]: will fix in later PR

**Pull request merged**